src/runtime_src/core/edge/drm/zocl/zocl_kds.c

```diff
@@ -414,13 +414,15 @@ int zocl_command_ioctl(struct drm_zocl_dev *zdev, void *data,
 		start_krnl_ecmd2xcmd(to_start_krnl_pkg(ecmd), xcmd);
 		break;
 	case ERT_EXEC_WRITE:
-		/* third argument in following function is representing number of
-		 * words to skip when configuring CU. This should be consistent
-		 * for both edge/DC, but due to performance and siome use cases,
-		 * this has veen decided that , DC flows skip 6 words whereas
-		 * edge flows doesnt skip any words.
-		 */
-		exec_write_ecmd2xcmd(to_start_krnl_pkg(ecmd), xcmd, 0);
+		DRM_WARN("ERT_EXEC_WRITE is obsoleted, use ERT_START_KEY_VAL\n");
```

Comment:
I am a little concerned about this breaking existing applications.  Do you think maybe we can shift the payload 6 words up (at performance impact of course, with the warning you have), then call start_krnl_kv_ecmd2xcmd?   The good news is that API level usage will automatically use ERT_START_KEY_VAL, it is only if someone out there is explicitly using ERT_EXEC_WRITE on an explicitly allocated exec buffer.

Reply:
On EDGE, we educated everyone that explicitly using ERT_EXEC_WRITE to not add that 6 words.
So, for zocl_kds.c, ERT_EXEC_WRITE is equal to ERT_START_KEY_VAL.

Reply:
For DC platforms, do you suggest to move the payload 6 words up?

Reply:
Sorry, I didn't see that I commented on the zocl version.  Yeah, I was suggesting moving xocl 6 words up.

Reply:
@stsoe update xrt++ and test with modified tests/unit_test/experimental/exec_write test case.
For xocl, if user still use ERT_EXEC_WRITE command, driver would convert it into ERT_START_KEY_VAL without considering performance impact.

I would add this change to 2021.2 release CHANGELOG after it is merged.